tests/components/demo/test_geo_location.py

```diff
@@ -62,5 +64,5 @@ def test_setup_platform(self):
             # Get all states again, ensure that the number of states is still
             # the same, but the lists are different.
             all_states_updated = self.hass.states.all()
-            assert len(all_states_updated) == NUMBER_OF_DEMO_DEVICES
+            assert len(all_states_updated) == NUMBER_OF_DEMO_DEVICES + 1
```

Comment:
@awarecan why is there a +1 needed here? I think that's causing the current test failure.

Reply:
`demo` integration has `zone` deps in the manifest.json. However, `demo.geo_location` doesn't have such dependency requirement. 

In current loading mechanism, only platform dependency will be check, domain component would not be load unless specific such dependency in platform file. 

It got changed in #22716, which will always check integration's dependency, not entity platform's.

I can roll this test back and we can discuss who to deal with this difference in other pull requests.